lib/autocomplete.ts

```diff
@@ -97,16 +97,18 @@ class Autocomplete extends BaseMenu implements IPopupContent {
         this.prevIndex();
         this._updateValue(trigger);
       },
-      // On Enter key, if an item is selected we update the value and close the dropdown. We always
-      // let event propagate (notice the '$' at the end of 'Enter$') because it is often desirable
-      // to listen to the enter key on an input element. Note that the items' action do not run on
-      // Enter because the focus remain on the trigger element.
+      // On Enter (and Tab) key, we update the trigger's value and close the dropdown. We always let
+      // event propagate (notice the '$' at the end of 'Enter$') because it is often desirable to
+      // listen to eigher one of Enter (or Tab) key on an input element. Note that the menu items'
+      // action does not run on Enter because the focus remains on the trigger element at all time.
       Enter$: (ev) => {
-        if (this._selected) {
-          trigger.value = this._selected.textContent!;
-          ctl.close();
-        }
+        this._updateValue(trigger);
```

Comment:
This is a change of behavior on Enter, that you didn't mention. Could you explain it?

Reply:
In practice it does not change the behavior. This change allow to remove some code duplication. 
 - I realized that what was needed to update the value of the trigger was already in `_updateElement`. 
 - When an item is selected, it sets the value to `this._selected.textContent!;` just like before.
 - when not item is selected, the value of the trigger is already that of `this._lastAsTyped` so no actual changes.
 - leaving menu open when nothing was selected seemed a mistake, now it closes always.